embedded-hal/src/mmc.rs

```diff
@@ -0,0 +1,101 @@
+//! Types and traits for SD/MMC peripherals.
+
+mod bus_width;
+mod fifo_status;
+mod reset;
+
+pub mod command;
+pub mod response;
+pub mod tuning;
+
+pub use bus_width::BusWidth;
+pub use fifo_status::FifoStatus;
+pub use reset::Reset;
+
+use command::MmcCommand;
+use response::MmcResponse;
+use tuning::TuningMode;
+
+/// Common operations for DesignWare MMC controllers on JH7110 SoCs.
+pub trait MmcOps {
+    /// Associated error type for the SD/MMC trait.
+    type Error;
+
+    /// Gets whether the device is a MMC card.
+    fn is_mmc(&self) -> bool;
```

Comment:
Imo all these bools should be enums, like `CardType::Sd` and `TransportMode::Spi`

Reply:
Alright, I also prefer new-types, so I'll make those changes.